### PR TITLE
WebXR API files, part I.

### PR DESCRIPTION
**Pull request by jpmedley**
@Elchi3 Here's the first part of the WebXR files that @a2sheppy told you about. There are 37 in all. I plan to submit them in batches to make the reviewing psychologically easier. I'll do the others after this one is merged as a way to sanity check what I'm doing.

### CODE REVIEW

File: api/Navigator.json [hunk 1]
```diff
@@ -2267,6 +2267,54 @@
             "deprecated": false
           }
         }
+      },
+      "xr": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Navigator/webdriver",
```
Comment (a2sheppy) [inline api/Navigator.json]: `webdriver` --> `xr`

File: api/XR.json [hunk 2]
```diff
@@ -0,0 +1,196 @@
+{
+  "api": {
+    "XR": {
+      "__compat": {
+        "mdn_url": "https://developer.mozilla.org/docs/Web/API/XR",
```
Comment (a2sheppy) [inline api/XR.json]: https://developer.mozilla.org/docs/Web/API/WebXR_API

**Comment by jpmedley** [comment]
Fixed. Please take another look.

**Comment by jpmedley** [comment]
The only thing I see is the navigator.xr fix.

**Comment by a2sheppy** [comment]
> The only thing I see is the navigator.xr fix.

Dammit, VS Code, your review mode bit me bad.

**Comment by a2sheppy** [comment]
@jpmedley 	In general, methods should have a `description` in the form `<code>methodName()</code>`, to get the parens that indicate it's a method. It's a workaround, but it's what we do these days.

The `XR` interface has `ondevicechange` which means it should also have an entry called `devicechange_event` linked to `.../XR/devicechange_event` and a description `<code>devicechange</code> event`.

In `XRFrameRequestCallback.json`, the feature name is accidentally `XRLayer` (which is never mentioned in the spec). on line 3.

And there are several things which don't seem to exist.

* The spec I'm looking at (dated July 22, 2019) does not have a `worldInformation` property in `XRFrame`.
* That same spec also does not have anything called `XRHitResult`.

**Comment by jpmedley** [comment]
Everything should be done. 

I've removed the stuff you can't find in the spec. I plan to confirm with the spec author that it is Chromium and not the spec that is out of date. I'll start keeping my eye out for discrepancies and check them individually.

File: api/XR.json [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        },
+        "status": {
+          "experimental": true,
+          "standard_track": false,
+          "deprecated": false
+        }
+      },
+      "devicechange_event": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/XR/ondevicechange",
```
Comment (Elchi3) [inline api/XR.json]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/XR/devicechange_event",
```

File: api/XR.json [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+          "description": "<code>devincechange</code> event",
```
Comment (Elchi3) [inline api/XR.json]:
```suggestion
          "description": "<code>devicechange</code> event",
```

**Comment by Elchi3** [comment]
> In general, methods should have a description in the form <code>methodName()</code>, to get the parens that indicate it's a method. It's a workaround, but it's what we do these days.

This isn't an official data convention. Please file a bcd issue so we can discuss and make it a convention in our data guidelines doc.

**Comment by jpmedley** [comment]
Changes committed. Please take another look.

**Comment by jpmedley** [comment]
@a2sheppy Is there anything else I need to do?

**Comment by Elchi3** [comment]
I can make the call here and get this in since I've already reviewed and approved this. The sub features aren't sorted alphabetically but that's not blocking since we're not enforcing that (yet).

**Comment by jpmedley** [comment]
Yikes! I thought I had sorted them.

**Comment by bershanskiy** [comment]
I'm confused about why `XR` and its attributes are labeled `"standard_track": false`... They are still in [the most recent draft I see](https://immersive-web.github.io/webxr/#xr-interface) and, in a way, `XR` attached to `navigator.xr` are necessary to use the whole API.
What am I missing?

**Comment by Elchi3** [comment]
Good catch, @bershanskiy :+1: 
I think  `"standard_track": false` was a mistake here. Sorry for not catching that in my review. @jpmedley can you comment on that or create a follow-up PR?

**Comment by bershanskiy** [comment]
Thanks for the prompt response, @Elchi3 
Also, what is the MDN's policy on which IDL definitions get their own pages and which do not? Specifically, I created a page for [`XRFrameRequest`](https://developer.mozilla.org/en-US/docs/Web/API/XRFrameRequestCallback) after I saw [XRFrameRequestCallback.json](https://github.com/mdn/browser-compat-data/blob/master/api/XRFrameRequestCallback.json) merged in this PR. However, it is just a callback used in only one function [`XRSession.requestAnimationFrame()`](https://developer.mozilla.org/en-US/docs/Web/API/XRSession/requestAnimationFrame) and defined as:
```
callback XRFrameRequestCallback = void (DOMHighResTimeStamp time, XRFrame frame);
```
I think browser support for a function `XRSession.requestAnimationFrame()` already implies support for a callback it takes. If this is the case, I don't think it is notable enough to have its own page.

**Comment by jpmedley** [comment]
I corrected the standards error in #4596

**Comment by Elchi3** [comment]
> Also, what is the MDN's policy on which IDL definitions get their own pages and which do not? 

There isn't currently a policy. I agree that it is probably not notable enough, however I'm not the primary author of the WebXR docs and I defer this question to the subject matter experts creating the material.

**Comment by jpmedley** [comment]
@bershanskiy How much time do you intend to devote to WebXR? I'm working on creating boilerplates for the WebXR interfaces&mdash;basic page structure with little more than interface and member defintions&mdash;so that @a2sheppy myself and others can add details. I'm worried that we'll be trying to create the same pages at the same times.

**Comment by jpmedley** [comment]
> > Also, what is the MDN's policy on which IDL definitions get their own pages and which do not?
> 
> There isn't currently a policy. I agree that it is probably not notable enough, however I'm not the primary author of the WebXR docs and I defer this question to the subject matter experts creating the material.

What you propose is consistent with [Window.requestAnimationFrame()](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) (and for what it's worth at this point, [VRDisplay.requestAnimationFrame()](https://developer.mozilla.org/en-US/docs/Web/API/VRDisplay/requestAnimationFrame).

**Comment by bershanskiy** [comment]
@Elchi3 
> I defer this question to the subject matter experts creating the material.

Do you happen to know someone we could @ mention? I decided to create that page (hesitantly) because this PR had a file `api/XRFrameRequestCallback.json` with a MDN URL set and didn't want to leave a broken URL in BCD table :). So far, I'm the only [contributor to it](https://developer.mozilla.org/en-US/docs/Web/API/XRFrameRequestCallback$history). I would prefer we delete `api/XRFrameRequestCallback.json` and the article itself.

**Comment by bershanskiy** [comment]
@jpmedley 
> How much time do you intend to devote to WebXR?

I don't know yet... I just had some free time and I know how crucial docs are for an API to gain traction. I just want to ensure WebXR docs are at least as god as those for WebVR so people can transition from WebVR to WebXR and start developing for WebXR. Also, I thought it would be fitting to get them done before the public launch (in Chrome 78, I believe). I think, that's your intention as well, right?

> I'm worried that we'll be trying to create the same pages at the same times.

Why would that be a problem? If you want, I can stop my activities. So far, I was making sure my MDN contributions stay consistent with your PRs here.

**Comment by bershanskiy** [comment]
> What you propose is consistent with Window.requestAnimationFrame() (and for what it's worth at this point, VRDisplay.requestAnimationFrame().

Good point, we should keep docs consistent. So far, neither of those has pages just for callbacks. However, what is more worrisome is all three call their callbacks `FrameRequestCallback` so creating file `api/FrameRequestCallback .json` is pretty ambiguous -- it can be mistaken for describing either of the three callbacks!

**Comment by a2sheppy** [comment]
For what it's worth, I decided a long while back that for ease of knowing where to link to when writing, I would give just about everything its own page, pretty much regardless of how simple or trivial. The exceptions have been really rare.

**Comment by a2sheppy** [comment]
@bershanskiy BTW, there are a number of issues with the structure and style of the pages you're adding to MDN right now. I tried to email you last week but it bounced. Now that I have a way to reach you, I'll tell you on GitHub. :)

**Comment by a2sheppy** [comment]
@bershanskiy Here's the text of the email I tried to send you about the documents you've been adding:

I’ve edited the article on XRFrame to serve as an example, but some of the specific points that I noticed:

- Methods always have “()” at the end, like “getUserMedia()”.
- The `{{domxref}}` macro used in the `<dt>` element in the lists of attributes and methods on interface overview pages should include the optional second parameter, specifying just the attribute or method name, so that instead of “XRFrame.session” and “XRFrame.getPose”, you’re seeing “session” and “getPose()” (this is how you add the parentheses, too).
- When adding the browser compatibility section, there’s a block of hidden content that provides a reminder to the writer while editing as to how the tables are now being created.
- We no longer use the {{SeeCompatTable}} macro on pages, even for experimental APIs. We do still use the {{experimental_inline}} on the items in the attribute and method lists, though.

**Comment by bershanskiy** [comment]
@a2sheppy Thank you for notifying me of this and I'm sorry about that. I'll go back and fix all macros on the pages I created.

So just to be clear: In general {{SeeCompatTable}} should be removed and, if applicable, {{experimental_inline}} should be added in the attributes and methods lists, right?

**Comment by a2sheppy** [comment]
> So just to be clear: In general {{SeeCompatTable}} should be removed and, if applicable, {{experimental_inline}} should be added in the attributes and methods lists, right?

So, personally, I wouldn't bother with the `{{experimental_inline}}` either, because the compatibility table should make it pretty clear. It's acceptable to include it, though.

**Comment by jpmedley** [comment]
These issues are exactly why I'm trying to work ahead of you on creating boilerplates.

> @bershanskiy Here's the text of the email I tried to send you about the documents you've been adding:
> 
> I’ve edited the article on XRFrame to serve as an example, but some of the specific points that I noticed:
> 
> * Methods always have “()” at the end, like “getUserMedia()”.
> * The `{{domxref}}` macro used in the `<dt>` element in the lists of attributes and methods on interface overview pages should include the optional second parameter, specifying just the attribute or method name, so that instead of “XRFrame.session” and “XRFrame.getPose”, you’re seeing “session” and “getPose()” (this is how you add the parentheses, too).
> * When adding the browser compatibility section, there’s a block of hidden content that provides a reminder to the writer while editing as to how the tables are now being created.
> * We no longer use the {{SeeCompatTable}} macro on pages, even for experimental APIs. We do still use the {{experimental_inline}} on the items in the attribute and method lists, though.

**Pull request merged**